apis.py

```diff
@@ -494,191 +492,275 @@ def update_volunteer_info(*args,**kwargs):
+        return json.dumps(
+            {'Response': {}, 'status': False, 'string_response': 'This page does not exist. Redirecting to homepage'})
     else:
-        return json.dumps({'Response':df.to_dict('records'),'status':True,'string_response':'Request related data extracted'})
+        return json.dumps(
+            {'Response': df.to_dict('records'), 'status': True, 'string_response': 'Request related data extracted'})
 
 
 # In[ ]:
+@app.route('/ngo_request_form', methods=['POST'])
+@login_required
+def ngo_request_form(*args, **kwargs):
+    name = request.form.get('name')
+    mob_number = request.form.get('mob_number')
+    email_id = request.form.get('email_id', '')
+    age = request.form.get('age', 70)
+    address = request.form.get('address')
+    geoaddress = request.form.get('geoaddress', address)
+    user_request = request.form.get('request')
+    latitude = request.form.get('latitude', 0.0)
+    longitude = request.form.get('longitude', 0.0)
+    source = request.form.get('source', 'covidsos')
+    status = request.form.get('status', 'received')
+    country = request.form.get('country', 'India')
+    uuid = generate_uuid()
+    what = request.form.get('geoaddress')
+    why = request.form.get('why')
+    financial_assistance = request.form.get('financial_assistance', 0)
+    verification_status = 'pending'
+    where = geoaddress
+    verified_by = kwargs.get('user_id', 0)
+    urgent_status = request.form.get('urgent', 'no')
+    volunteers_reqd = request.form.get('volunteer_count', 1)
+    current_time = dt.datetime.utcnow() + dt.timedelta(minutes=330)
+    req_dict = {'timestamp': [current_time], 'name': [name], 'mob_number': [mob_number], 'email_id': [email_id],
+                'country': [country], 'address': [address], 'geoaddress': [geoaddress], 'latitude': [latitude],
+                'longitude': [longitude],
+                'source': [source], 'age': [age], 'request': [user_request], 'status': [status], 'uuid': [uuid],
+                'volunteers_reqd': [volunteers_reqd]}
+    df = pd.DataFrame(req_dict)
+    df['email_id'] = df['email_id'].fillna('')
+    expected_columns = ['timestamp', 'name', 'mob_number', 'email_id', 'country', 'address', 'geoaddress', 'latitude',
+                        'longitude', 'source', 'request', 'age', 'status', 'uuid']
+    x1, y1 = add_requests(df[expected_columns])
+    r_df = request_data_by_uuid(uuid)
+    r_v_dict = {'r_id': [r_df.loc[0, 'id']], 'why': [why], 'what': [what], 'where': [where],
+                'verification_status': [verification_status], 'verified_by': [verified_by], 'timestamp': [current_time],
+                'financial_assistance': [financial_assistance], 'urgent': [urgent_status]}
+    df = pd.DataFrame(r_v_dict)
+    expected_columns = ['timestamp', 'r_id', 'what', 'why', 'where', 'verification_status', 'verified_by',
+                        'financial_assistance', 'urgent']
+    x2, y2 = add_request_verification_db(df[expected_columns])
```

Comment:
this is the only difference between a non ngo request and an ngo request? Not sure I understand what that is meant for.

Reply:
An NGO request form is both request form and verification form combined. This is to be filled by a person representing an NGO. That ways, our team centrally just has to approve it.